// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.a.stwo";
option java_outer_classname = "stwo";
option objc_class_prefix = "SWTO";

package STwo;

service ServiceTwo {
  rpc GetProducts(ProductsRequest) returns (ProductsResponse) {}
}

message ProductsRequest{

}

message ProductsResponse {
  repeated Product listProducts = 1;
}

message Product{
    string Id= 1;
    int32 price_in_cents = 2;
    string title = 3;
    string description = 4;
	Discount discount = 5;
}

message Discount {
        float pct = 1;
        int32 value_in_cents = 2;
}

